src/fixSuggestions/fixSuggestionsService.ts

```diff
@@ -0,0 +1,46 @@
+import { ShowFixSuggestionParams } from "../lsp/protocol";
+import * as vscode from "vscode";
+import { logToSonarLintOutput } from "../util/logging";
+import { SonarLintExtendedLanguageClient } from "../lsp/client";
+
+export class FixSuggestionService {
+	private static readonly END_OF_LINE_OFFSET = 10_000;
+	private static _instance : FixSuggestionService;
+
+	static init(client: SonarLintExtendedLanguageClient) {
+		FixSuggestionService._instance = new FixSuggestionService(client);
+	}
+	constructor(private readonly client: SonarLintExtendedLanguageClient) {}
+
+	static get instance() {
+		return FixSuggestionService._instance;
+	}
+
+	showFixSuggestion = async (params : ShowFixSuggestionParams) => {
+		try {
+			const fileUri = vscode.Uri.parse(params.fileUri);
+			await vscode.window.showTextDocument(fileUri);
+			const wsedit = new vscode.WorkspaceEdit();
+			params.textEdits.forEach(async (edit) => {
+				const range = new vscode.Range(edit.beforeLineRange.startLine, 0, edit.beforeLineRange.endLine, FixSuggestionService.END_OF_LINE_OFFSET);
```

Comment:
What if the start/end lines are wrong? (i.e. outside the bounds of the document)

Reply:
Good point!! I just pushed a commit to apply change to a "validated range". For example, if we receive a range that is completely out of the file, like line 100 for a 5-line file, the user will see a warning message that "file content has changed and fix might not be applicable anymore", and they will see a fix suggestion at the very end of the file.